js/ui/overview.js

```diff
@@ -432,9 +432,18 @@ var Overview = class {
         // To avoid updating the position and size of the workspaces
         // we just hide the overview. The positions will be updated
         // when it is next shown.
+        const visible = this.visible;
         this.hide();
 
         this._relayoutNoHide();
+
+        // The previous hide call will make *NotVisible* the apps icons
+        // in the desktop. If it was visible before this relayout we should
+        // show again to avoid a desktop with disabled icons and no apps opened
+        if (visible) {
+            this.visible = false;
+            this.showApps();
+        }
```

Comment:
What causes the call to relayout() when the laptop lid is closed? Is there any way to avoid that?

I guess this also causes the apps page to be displayed if the laptop lid is closed while showing the window picker? Perhaps we should try to redisplay whichever page was visible, if we can't avoid the call to relayout at all.

Reply:
The relayout is because the lid is closed. This callback is triggered by the `monitors-changed` signal, emitted in the `LayoutManager` class and that signal is directly connected to the mutter `MonitorManager`.